### ISSUE TITLE
Response getters hasState() and HasSessionData() not working for google responses

### ISSUE THREAD

**Issue by natrixx** [jovotech/jovo-framework#462]
## I'm submitting a...

- [X ] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Current Behavior
While working on some google tests I noticed that response getters `hasSessionData()` and `hasState()` are returning `true` no matter what. I found the lines of code below that appear to be causing the issue. The getters work fine for Alexa testing.

### Error log
https://github.com/jovotech/jovo-framework/blob/7ff6f7c73b57994398a035f51b89b388dc2c2096/jovo-integrations/jovo-platform-googleassistant/src/core/GoogleActionResponse.ts#L82-L97

## Your Environment

* Jovo Framework version used: 2.2.0
* Operating System: Ubuntu

**Comment by aswetlow** [comment]
Hey @natrixx,

I tried to reproduce with this [sample code](https://github.com/jovotech/jovo-framework/blob/840543c6e9834c55dd733c70a6e4b70b4a4ce7d7/examples/javascript/unit-testing/test/sample.test.js#L7) with some modifications. Without success.

```javascript
 LAUNCH() {
       return this.toIntent('HelloWorldIntent');
    },

    HelloWorldIntent() {
        this.$session.$data.hello = 'world';
        this.followUpState('IntroductionState')
            .ask('Hello World! What\'s your name?', 'Please tell me your name.');
    },
```
Test
```javascript
 test('should return a welcome message and ask for the name at "LAUNCH"', async () => {
            const conversation = testSuite.conversation();

            const launchRequest = await testSuite.requestBuilder.launch();

            const responseLaunchRequest = await conversation.send(launchRequest);

            expect(
                responseLaunchRequest.isAsk('Hello World! What\'s your name?', 'Please tell me your name.')
            ).toBe(true);


            expect(
                responseLaunchRequest.hasState('IntroductionState')
            ).toBe(true);


            expect(
                responseLaunchRequest.hasSessionData('hello')
            ).toBe(true);
        });
```
Could you post your code, please?

The GoogleActionResponse doesn't have information about the session. The `hasState`, `hasSessionData` logic is in `DialogflowResponse.ts`

**Comment by natrixx** [comment]
Thanks @aswetlow ! 

In your examples, you are expecting back true data. Also using the sample code you did, I updated to the following:

First, remove the followup state to keep the request stateless (my code is also testing to make sure no state is set)
```
 LAUNCH() {
        return this.toIntent('HelloWorldIntent');
    },

    HelloWorldIntent() {
      this.$session.$data.hello = 'world';
      this.ask('Hello World! What\'s your name?', 'Please tell me your name.');
  },
```

changed the last two of your expectation blocks to the following:

```
test('should return a welcome message and ask for the name at "LAUNCH"', async () => {
            const conversation = testSuite.conversation();

            const launchRequest = await testSuite.requestBuilder.launch();

            const responseLaunchRequest = await conversation.send(launchRequest);

            expect(
                responseLaunchRequest.isAsk('Hello World! What\'s your name?', 'Please tell me your name.')
            ).toBe(true);

            expect(
                responseLaunchRequest.hasState()
            ).toBe(false);

            expect(
                responseLaunchRequest.hasSessionData('hello', 'world123')
            ).toBe(false);
        });
```

When ran, `hasState` should pass for Alexa, then fail for Google. 

Then comment out the `hasState` check and run again. Should get the same results as above, Alexa test passes, but google fails.

**Comment by natrixx** [comment]
After some digging around today, I found that this line here appeared to be an issue for the `hasSessionData` error above:

https://github.com/jovotech/jovo-framework/blob/master/jovo-integrations/jovo-platform-dialogflow/src/core/DialogflowResponse.ts#L170

it seems that for google, only the existence of the attribute is checked, with the value never being checked. So in my example above, I get a true response as hello exists. However for alexa, both are checked. 

So in the end, I asked myself "why are these methods not setup the same to begin with?" I changed the google code to match the alexa code and now the above tests and anything else I test appear to work as desired and fairly bullet proof. 

So I have created my very first Jovo PR (hopefully not my last! ) to address and discuss further!
